src/test.rs

```diff
@@ -110,3 +110,25 @@ Pick three.";
         vec!["Safe, fast, productive."]
     );
 }
+
+#[test]
+fn stdin_as_input() {
```

Comment:
I don't believe this text is doing what is expected.

The functionality added on this PR is: *read text from stdin*. A test for that functionality, I believe, should actually read text from the standard input.

The test should call the `run` function with no `filename` defined on the `Config` struct and assert the function behaves as expected. In this case, expected behaviour for the `run` function would be printing to `stdout`.

What the test does is assert the function `search_case_insensitive` works (the same thing the function `case_insensitive` does). Am I mistaken?

Reply:
In this case yes, it asserts that search_case_insensitive works.  
I will redo the test so it has usefulness.

Reply:
Take a look at the `base_program_run` test.

One possible approach is to refactor that test: make a function that mimics its behaviour, calling the `run` function and use that function on the `base_program_run` and on the `stdin_as_input` tests.